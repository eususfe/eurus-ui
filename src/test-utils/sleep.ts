/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
