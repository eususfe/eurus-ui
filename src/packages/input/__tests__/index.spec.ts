import { mount } from '@vue/test-utils';
import EInput from '..';

describe('EInput', () => {

  it('EInput snapshot', () => {
    const wrapper = mount(EInput);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should  emit input event', async () => {
    const fn = vitest.fn();
    const wrapper = mount(EInput, {
      props: {
        value: 'Test',
        onInput: fn
      }
    });
    const input = wrapper.find('input');
    input.element.value = 'Test2';
    await input.trigger('input');

    expect(fn).toHaveBeenCalled();

    wrapper.unmount();

  });

  it('should work with clearable prop', async () => {
    const fn = vitest.fn();
    const wrapper = mount(EInput, {
      props: {
        'modelValue': 'test',
        'clearable': true,
        'onUpdate:modelValue': fn
      }
    });

    await wrapper.find('.eu-input-clearable').trigger('click');
    expect(fn).toHaveBeenCalled();
    wrapper.unmount();
  });
  it('should work with disable prop', () => {
    const wrapper = mount(EInput, {
      props: {
        disabled: true
      }
    });
    expect(wrapper.find('.eu-input-wrapper--disabled').exists()).toBe(true);
    wrapper.unmount();
  });

  it('should work with size prop', () => {
    const size = ['xs', 'sm', 'md', 'lg', 'xl'] as const;
    size.forEach((size) => {
      const wrapper = mount(EInput, {
        props: {
          size
        }
      });
      expect(wrapper.find('.eu-input-wrapper--' + size).exists()).toBe(true);
      wrapper.unmount();
    }
    );
  });
  it('should work with showWordLimit and maxLength prop', async () => {
    const wrapper = mount(EInput, {
      props: {
        modelValue: '12345678901',
        showWordLimit: true,
        maxLength: 10
      }
    });

    expect(wrapper.find('input').attributes('maxlength')).toBe('10');
    expect(wrapper.find('.eu-input-suffix').find('.eu-input-word-limit').exists()).toBe(true);
  });
});
