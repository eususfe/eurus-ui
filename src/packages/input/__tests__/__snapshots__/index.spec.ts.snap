// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EInput > EInput snapshot 1`] = `"<span class=\\"eu-input-wrapper eu-input-wrapper--md\\"><!----><input type=\\"text\\" tabindex=\\"0\\" placeholder=\\"\\" class=\\"eu-input\\"><!----><!----></span>"`;

exports[`eInput > eInput snapshot 1`] = `"<span class="eu-input-wrapper eu-input-wrapper--md"><!----><input type="text" tabindex="0" placeholder="" class="eu-input"><!----><!----></span>"`;
